src/xinterpreter.cpp

```diff
@@ -327,16 +327,8 @@ namespace xcpp
         // Attempt normal evaluation
         
         try
-        {   std::string exp = R"(\w*(?:\:{2}|\<.*\>|\(.*\)|\[.*\])?)";
-            std::regex re(R"((\w*(?:\:{2}|\<.*\>|\(.*\)|\[.*\])?)(\.?)*$)");
-            auto inspect_request = is_inspect_request(code, re);
-            if (inspect_request.first)
-            {
-                inspect(inspect_request.second[0], kernel_res, *m_interpreter);
-            }
-            
+        {
```

Comment:
The `is_inspect_request` needs to parse the code, before `process_code` and catch the cases where we have a request such as: `?std::vector`

The [`no documentation printout`](https://github.com/compiler-research/xeus-cpp/blob/7c7d42de1cc40b595418d8a675c6648ea7dc1860/src/xinspect.hpp#L220) could be fixed with a better regex that only matches when we have a `?` to check for documentation

Reply:
Ohh so an execution request and an inspect request aren't decoupled. As in if I want to inspect `?std::vector` , shouldn't that be calling the `inspect_request_impl` ? 

I thought that something like the above would not end up calling `execute_request_impl`. I don't see close relatives of xeus-cpp (like xeus-cling and xeus-clang-repl) indulge in any kind of inspection in the execute request code (https://github.com/jupyter-xeus/xeus-cling/blob/471b1b68e3cad35746652cf14ce40ce0b7797ba0/src/xinterpreter.cpp#L125, https://github.com/compiler-research/xeus-clang-repl/blob/54d7f2c1d373855525ab2e9b86da3164376d7e1f/src/xinterpreter.cpp#L113) 

Could you explain this to me ? 
cc @ioanaif